packages/cli/src/lib/decorators/preconditions/apiKeyPrivilege.ts

```diff
@@ -0,0 +1,93 @@
+import PlatformClient, {
+  PrivilegeEvaluatorModel,
+} from '@coveord/platform-client';
+import Command from '@oclif/command';
+import dedent from 'ts-dedent';
+import {Config} from '../../config/config';
+import {AuthenticatedClient} from '../../platform/authenticatedClient';
+
+export const impersonatePrivilege = {
+  requestedPrivilege: {
+    owner: 'SEARCH_API',
+    targetDomain: 'IMPERSONATE',
+    targetId: '*',
+  },
+};
+
+export const createApiKeyPrivilege = {
+  requestedPrivilege: {
+    owner: 'PLATFORM',
+    targetDomain: 'API_KEY',
+    targetId: '*',
+    type: 'CREATE',
+  },
+};
+
+export function HasNecessaryCoveoPrivileges() {
+  return async function (target: Command) {
+    const authenticatedClient = new AuthenticatedClient();
+
+    const client = await authenticatedClient.getClient();
+    const {organization, anonymous} = await getConfiguration(target);
+
+    console.log('*********************');
+    console.log(anonymous);
+    console.log('*********************');
+
+    if (!(await hasCreateApiKeyPrivilege(client, organization))) {
+      target.warn(
+        anonymous
+          ? 'Your API key is missing the privilege to create other API keys. Make sure to grant this privilege before running the command again.'
+          : 'You are not authorized to create an API Key. Please contact an administrator of your Coveo organization.'
```

Comment:
Could we link to some article on docs.coveo.com that explains the concept better ?